build.md

```diff
@@ -227,6 +227,34 @@ Cache target is defined using the same `type=TYPE[,KEY=VALUE]` syntax defined by
 
 Unsupported cache target MUST be ignored and not prevent user from building image.
 
+### additional_contexts
+
+`additional_contexts` defines a list of named contexts the image builder SHOULD use during image build.
```

Comment:
I think we should be aligned with current syntax buildx to avoid confusion: https://github.com/docker/buildx/blob/master/docs/reference/buildx_build.md#-additional-build-contexts---build-context

So I would say `build_contexts` would be better or even better be aligned with [bake specification](https://docs.docker.com/build/bake/file-definition/#specification) so it would be `contexts`.

Reply:
we had this discussion here too https://github.com/compose-spec/compose-spec/pull/307#discussion_r1110116993, and here https://github.com/compose-spec/compose-spec/issues/306. I'd be okay with `build_contexts` too (as was my initial proposal), but I feel like `contexts` is too similar to the already existing `context`. cc @ndeloof @milas, maybe we can try to get all the discussion in one place 😅

Reply:
sounds weird to me to define `build_contexts` inside the `build` section.
the buildx docs mentioned ^ use `addtional-(build)-contextx`, so my suggestion

Reply:
fwiw `named_contexts` is also an option here

Reply:
Agree with @ndeloof , we're already in a `build` section and the `context` attribute already defines the default context, so `build_contexts` is not the best option here. 
`additional_contexts` or `named_contexts` seem to be better suitable names, even if I prefer the 1st one  which is more meaningful

Reply:
If most of us agree, maybe we can keep `additional_contexts` then :)

Reply:
I have a preference towards `contexts` for existing consistency with bake, but it seems as though that's not the direction this is headed. It feels frustrating from a user-experience though that we'd have different keys for different scenarios: the command line and build-push-action (`--build-context` and `build-contexts`), bake (`contexts`) and compose (`named_contexts`/`additional_contexts`). I'm tempted to propose consistency with command line options that users can already use today with buildx and other build tools like buildah - either `contexts` or `build-contexts` sound good to me for this. 

That said, I don't want to stall the discussion, I'm alright with `additional_contexts` and `build_contexts` if there's a strong majority here.

Reply:
@EricHripko any thoughts on this one?

Reply:
Having `build.build_contexts` seems a bit repetitive - it's already under `build` section, so `build_` prefix seems excessive. If we're really keen to keep consistent naming, then I would suggest that `build.context` acquires a long(ish?) form syntax - something akin to:
```yaml
# Single context
context: .
context: path/to/dir

# Multiple contexts
context:
  - .
  - alpine=docker-image://alpine@sha256:0123456789
context:
  - path/to/dir
  - project=path/to/project/source
```
From user perspective, the split between primary and additional build contexts seems like a bit of implementation details leaking into user space. Essentially, I'd consider all inputs to my build to be my build context.

In my (rather anecdotal) experience, I've seen very few developers be familiar with the newer `docker buildx` CLI and even fewer then heard/used bake in any shape or form. From that perspective, I am more keen on Compose spec making sense/being user-friendly as I'd expect the majority of developers to exclusively reside in this world.

If we have to introduce a new attribute for this, then I guess I could live with `additional_contexts`.

Reply:
I agree re: buildx and bakefile syntax, generally consistency is good but in this case I'd prioritize readability/clarity of the spec vs. strict adherence to the other syntaxes.

Re: multiple contexts in the `context` key though, I hadn't though about that! I'm okay with it, I only wonder if it's a bit confusing having a list where one unnamed entry is required and the (optional) others must be named – but I'm open to it if others agree with it.

Reply:
I agree. This would be fine if we didn't had to manage this distinction between the (required) main context and the named ones. But here I wonder this will bring some confusion, so I prefer we introduce an explicit `named_contexts` field

Reply:
Does the default/required context also have a name? BuildKit docs seem to suggest it's named `context`: `buildctl build --frontend=dockerfile.v0 --local context=. --local dockerfile=.`